corehq/apps/users/views/mobile/users.py

```diff
@@ -462,14 +463,15 @@ def delete_commcare_user(request, domain, user_id):
     user = CommCareUser.get_by_user_id(user_id, domain)
     if not _can_edit_workers_location(request.couch_user, user):
         raise PermissionDenied()
-    if (user.user_location_id and
-            SQLLocation.objects.get_or_None(location_id=user.user_location_id,
-                                            user_id=user._id)):
+
+    user_location = SQLLocation.objects.get_or_None(location_id=user.user_location_id, user_id=user._id)
+    if (user.user_location_id and user_location):
```

Comment:
Same as https://github.com/dimagi/commcare-hq/pull/33184/files#r1250332098

Reply:
also in this case the first clause in the if statement is a guard against the second - that is, "don't try to look up the location ID if it's `None`.  This change makes the lookup happen before the guard

Reply:
@sravfeyn Same reason for the refactor here as mentioned in the previous comment.

Reply:
https://github.com/dimagi/commcare-hq/pull/33184/commits/b1626985200778532e8c63e8d44f5af95e9c5770. In order to address @esoergel's comment, I had to switch out the `user_location` for `user_location_id`. Technically we could throw away the guard statement, since the query will return `None` if `user.location_id` is `None`, but I thought to keep it in, since it saves us the time of querying the DB now. "We value the time of our customers" :smile: